image-diff

```diff
@@ -0,0 +1,66 @@
+#!/usr/bin/python3
+
+# This file is part of Cockpit.
+#
+# Copyright (C) 2020 Red Hat, Inc.
+#
+# Cockpit is free software; you can redistribute it and/or modify it
+# under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation; either version 2.1 of the License, or
+# (at your option) any later version.
+#
+# Cockpit is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+
+import os
+import subprocess
+import sys
+
+def get_packages(image):
+    if any(image.startswith(prefix) for prefix in ['rhel', 'fedora', 'centos']):
+        # TODO: want source packages, but SOURCERPM is not great and the others don't work
+        pkgcmd = "rpm -qa --qf 'PACKAGE\t%{NAME}\t%{EVR}\n'"
+    elif any(image.startswith(prefix) for prefix in ['debian', 'ubuntu']):
+        pkgcmd = "dpkg-query -W -f='PACKAGE\t${source:Package}\t${version}\n'"
+    else:
+        sys.exit("Sorry, don't know how to list packages on " + image)
+
+    vm_run = os.path.join(os.path.dirname(__file__), 'vm-run')
+    output = subprocess.check_output([vm_run, '-e', pkgcmd, '-e', 'poweroff -f', image])
```

Comment:
This is already a Python script, so it'd be cleaner and easier to use the machines API directly. Check out machine/testvm.py and machine/machine_core/cli.py or https://cockpit-project.org/blog/cockpit-custom-test-framework.html about the details. No need to go via a cumbersome CLI and parsing stdout for these.